src/test/java/bio/terra/common/iam/AuthenticatedUserRequestTest.java

```diff
@@ -0,0 +1,158 @@
+          .registerModule(new JsonNullableModule())
+          .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+          .enableDefaultTyping(ObjectMapper.DefaultTyping.NON_FINAL);
+
+  private static final String EmailAddress = "test@example.com";
+  private static final UUID RequestId = UUID.randomUUID();
+  private static final String SubjectId = "Subject";
+  private static final String Token = "0123.456-789AbCd";
+
+  @Test
+  public void fullBuilder() throws Exception {
+    AuthenticatedUserRequest req =
+        AuthenticatedUserRequest.builder()
+            .setEmail(EmailAddress)
+            .setReqId(RequestId)
+            .setSubjectId(SubjectId)
+            .setToken(Token)
+            .build();
+
+    assertEquals(EmailAddress, req.getEmail());
+    assertEquals(SubjectId, req.getSubjectId());
+    assertEquals(Token, req.getToken());
+    assertEquals(Token, req.getRequiredToken());
+    assertNotNull(req.getReqId());
+    validateJsonSerialization(req);
+  }
+
+  @Test
+  public void emptyBuilder() throws Exception {
+    AuthenticatedUserRequest req = AuthenticatedUserRequest.builder().build();
+    assertNull(req.getEmail());
+    assertNull(req.getSubjectId());
+    assertNull(req.getToken());
+    assertNotNull(req.getReqId());
+    assertThrows(ApiException.class, req::getRequiredToken);
+    validateJsonSerialization(req);
+  }
+
+  @Test
+  public void equality() {
+    AuthenticatedUserRequest req =
+        AuthenticatedUserRequest.builder()
+            .setEmail(EmailAddress)
+            .setReqId(RequestId)
+            .setSubjectId(SubjectId)
+            .setToken(Token)
+            .build();
+
+    // Positive test
+    AuthenticatedUserRequest cmp = req.toBuilder().build();
+    assertEquals(req, cmp);
+    assertEquals(req.hashCode(), cmp.hashCode());
+
+    // Negative tests
+    assertNotEquals(req, req.toBuilder().setEmail("JUNK").build());
+    assertNotEquals(req, req.toBuilder().setReqId(UUID.randomUUID()).build());
+    assertNotEquals(req, req.toBuilder().setSubjectId("JUNK").build());
+    assertNotEquals(req, req.toBuilder().setToken("JUNK").build());
+
+    // Explicit test for comparison to self
+    assertEquals(req, req);
+
+    // Explicit test for off-type comparison
+    assertNotEquals(req, "test");
+  }
+
+  static class TestVector {
+    private final String json;
+    private final AuthenticatedUserRequest request;
+
+    public TestVector(String json, AuthenticatedUserRequest request) {
+      this.json = json;
+      this.request = request;
+    }
+
+    public void validate() throws JsonProcessingException {
+      AuthenticatedUserRequest deserialized =
+          objectMapper.readValue(this.json, AuthenticatedUserRequest.class);
+      assertEquals(request, deserialized);
+      assertEquals(request.hashCode(), deserialized.hashCode());
+    }
+  }
+
+  void validateJsonSerialization(AuthenticatedUserRequest request) throws JsonProcessingException {
```

Comment:
nit: this helper and the static class can both be private. Always make each class or member as inaccessible as possible, [ref](go/java-practices/visibility).